File: 3514-number-of-unique-xor-triplets-ii/3514-number-of-unique-xor-triplets-ii.cpp
class Solution {
public:
    int uniqueXorTriplets(vector<int>& nums) {
        unordered_set <int> unique_set(nums.begin() , nums.end());
        vector<int>unique_nums(unique_set.begin() , unique_set.end());

        int m = unique_nums.size();

        unordered_set<int> pair_xors;

        for(int i = 0 ; i < m ; i++){
            for(int j = i ; j < m ; j++){
                pair_xors.insert(unique_nums[i]^ unique_nums[j]);
            }
        }

        unordered_set<int> triplet_xors;
        for(int px : pair_xors){
            for(int num : unique_nums){
                triplet_xors.insert(px ^ num);
            }
        }

        return triplet_xors.size();
    }
};

File: 3514-number-of-unique-xor-triplets-ii/3514-number-of-unique-xor-triplets-ii.py
class Solution:
    def uniqueXorTriplets(self, nums: List[int]) -> int:
        unique_nums = list(set(nums))

        m = len(unique_nums)

        pair_xors = set()
        for i in range(m):
            for j in range(i , m):
                pair_xors.add(unique_nums[i] ^ unique_nums[j])

        triplet_xors = set()

        for px in pair_xors:
            for num in unique_nums:
                triplet_xors.add(px ^ num)
        
        return len(triplet_xors)